app/settings/data-export/data-export.controller.js

```diff
@@ -0,0 +1,102 @@
+module.exports = [
+    '$scope',
+    '$rootScope',
+    '$location',
+    'DataExport',
+    'Notify',
+    'FormEndpoint',
+    'FormAttributeEndpoint',
+    '$q',
+    '_',
+    'LoadingProgress',
+function (
+    $scope,
+    $rootScope,
+    $location,
+    DataExport,
+    Notify,
+    FormEndpoint,
+    FormAttributeEndpoint,
+    $q,
+    _,
+    LoadingProgress
+) {
+    $scope.exportAll = exportAll;
+    $scope.showFields = false;
+    $scope.showProgress = false;
+    $scope.selectFields = selectFields;
+    $scope.selectedFields = [];
+    $scope.exportSelected = exportSelected;
+    $scope.selectAll = selectAll;
+    $scope.displayedForms = [];
+    $scope.getForms = getForms;
+    $scope.isLoading = LoadingProgress.getLoadingState;
+    $scope.attachAttributes = attachAttributes;
+
+    // hiding export-in-progress screen when export is complete
+    $rootScope.$on('event:data_export:complete', function () {
+        $scope.showProgress = false;
+    });
+
+    // Redirect to home if not authorized
+    if ($rootScope.hasPermission('Bulk Data Import') === false) {
```

Comment:
Should we provide some kind of message if the user isn't authorized, rather than just redirecting?

Reply:
maybe, they should not really end up here if not authorised so its just a safety-check...